### PR TITLE
fix quad input hatch for Mega Oil Cracker

### PR DESCRIPTION
**Pull request by lc-1337**
I just copied code that check input hatch from regular oil cracker.

### CODE REVIEW

**Comment by lc-1337** [comment]
close https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/13809

**Comment by Pxx500** [comment]
Well, I'd say that oil cracker shouldn't support multiple(quad etc.) input hatches, after all it's a machine that's supposed to do one job on one fluid, just like distillery

**Comment by lc-1337** [comment]
Regular oil cracker support quad hatch, so why mega doesnt

**Comment by Pxx500** [comment]
> Regular oil cracker support quad hatch, so why mega doesnt

Well in that case I think the regular shouldn't too

**Comment by Pxx500** [comment]
@lordIcocain are you talking about inputing steam with the fuel or multiple fuels?

**Comment by lc-1337** [comment]
From quad hatch multiple fuels/radox, steam ofc from middle one.

**Comment by lc-1337** [comment]
There 2 main reason to use mega with multiple input, oil processing and radox processing.
In oil case building 3 mega looks weird, when you need so much stuff from oil, you gonna choose to get it from alternative source.
In radox case cracker not a bottleneck, when you overclock other machine your 4 regular oil cracker will work on 4x/8x less voltage, and then qft.
In my opinion nobody gonna use mega oil with only one input. Even now only few ppl use it.

**Comment by Pxx500** [comment]
Well, quadruple hatch sounds weird in oil cracker but I don't really have any balance wise arguments so I guess it's fine

**Comment by lc-1337** [comment]
bump

File: src/main/java/com/github/bartimaeusnek/bartworks/common/tileentities/multis/mega/GT_TileEntity_MegaOilCracker.java [hunk 1]
```diff
@@ -325,21 +326,40 @@ private boolean addMiddleInputToMachineList(IGregTechTileEntity aTileEntity, int
...
+            if (tHatch instanceof GT_MetaTileEntity_Hatch_MultiInput) {
+                if (isValidMetaTileEntity(tHatch)) {
+                    for (final FluidStack tFluid : ((GT_MetaTileEntity_Hatch_MultiInput) tHatch).getStoredFluid()) {
+                        if (tFluid != null && GT_Recipe.GT_Recipe_Map.sCrackingRecipes.isValidCatalystFluid(tFluid)) {
+                            rList.add(tFluid);
+                        }
+                    }
+                }
+            } else {
+                if (isValidMetaTileEntity(tHatch) && tHatch.getFillableStack() != null) {
```
Comment (BlueWeabo) [inline src/main/java/com/github/bartimaeusnek/bartworks/common/tileentities/multis/mega/GT_TileEntity_MegaOilCracker.java]: why not put the isValidMetaTileEntity before the instanceof check?

Comment (lc-1337) [reply src/main/java/com/github/bartimaeusnek/bartworks/common/tileentities/multis/mega/GT_TileEntity_MegaOilCracker.java]: Yes, it's make case when "isValidMetaTileEntity(tHatch)=false" faster, but in other cases there no difference.
So as you want.

Comment (BlueWeabo) [reply src/main/java/com/github/bartimaeusnek/bartworks/common/tileentities/multis/mega/GT_TileEntity_MegaOilCracker.java]: It makes it more readable. can go with: `if (!isValidMetaTileEntity(tHatch)) continue;`
at the start of the for loop.

Comment (lc-1337) [reply src/main/java/com/github/bartimaeusnek/bartworks/common/tileentities/multis/mega/GT_TileEntity_MegaOilCracker.java]: like a this?

**Review by BlueWeabo — APPROVED** [review]
looks good.

**Review by Dream-Master — APPROVED** [review]

**Pull request merged**